package/MDAnalysis/coordinates/RDKit.py

```diff
@@ -95,10 +134,591 @@ def __init__(self, filename, **kwargs):
+                "converter requires all hydrogens to be explicit. Please "
+                "check carefully the output molecule as the converter is "
+                "likely to add negative charges and assign incorrect bond "
+                "orders to structures with implicit hydrogens. Alternatively, "
+                "you can use the parameter `NoImplicit=False` when using the "
+                "converter to allow implicit hydrogens and disable inferring "
+                "bond orders and charges."
+            )
+
+        # attributes accepted in PDBResidueInfo object
+        pdb_attrs = {}
+        if hasattr(ag, "bfactors") and hasattr(ag, "tempfactors"):
+            raise AttributeError(
+                "Both `tempfactors` and `bfactors` attributes are present but "
+                "only one can be assigned to the RDKit molecule. Please "
+                "delete the unnecessary one and retry."
+            )
+        for attr in RDATTRIBUTES.keys():
+            if hasattr(ag, attr):
+                pdb_attrs[attr] = getattr(ag, attr)
+
+        other_attrs = {}
+        for attr in ["charges", "segids", "types"]:
+            if hasattr(ag, attr):
+                other_attrs[attr] = getattr(ag, attr)
+
+        mol = Chem.RWMol()
+        # map index in universe to index in mol
+        atom_mapper = {}
+
+        for i, (atom, element) in enumerate(zip(ag, elements)):
+            # create atom
+            rdatom = Chem.Atom(element.capitalize())
+            # disable adding H to the molecule
+            rdatom.SetNoImplicit(NoImplicit)
+            # add PDB-like properties
+            mi = Chem.AtomPDBResidueInfo()
+            for attr, values in pdb_attrs.items():
+                _add_mda_attr_to_rdkit(attr, values[i], mi)
+            rdatom.SetMonomerInfo(mi)
+            # other properties
+            for attr in other_attrs.keys():
+                value = other_attrs[attr][i]
+                attr = "_MDAnalysis_%s" % _TOPOLOGY_ATTRS[attr].singular
+                _set_atom_property(rdatom, attr, value)
+            _set_atom_property(rdatom, "_MDAnalysis_index", i)
+            # add atom
+            index = mol.AddAtom(rdatom)
+            atom_mapper[atom.ix] = index
+
+        try:
+            ag.bonds
+        except NoDataError:
+            warnings.warn(
+                "No `bonds` attribute in this AtomGroup. Guessing bonds based "
+                "on atoms coordinates")
+            ag.guess_bonds()
+
+        terminal_atom_indices = []
+        for bond in ag.bonds:
+            try:
+                bond_indices = [atom_mapper[i] for i in bond.indices]
+            except KeyError:
+                # one of the atoms of the bond is not part of the atomgroup.
+                # can happen for terminal atoms.
+                # save the bond atom that is in the atomgroup for later
+                terminal_atom_indices.extend([atom_mapper[i]
```

Comment:
Note, my understanding here is that this heavily relies on atomgroup_intersection acting the way it currently does. If not already doing so, we explicitly test terminal atom indices somehow to make sure that any future changes (I think it was mentioned somewhere), don't unintentionally prevent this from working properly.

Reply:
With the API change discussed in #2821 the try except won't be needed anymore.
Regarding the terminal_atom_indices, I was using it to avoid adding charges to these atoms, but standardize_patterns should be able to cover this now so I can simplify it a bit